go/pkg/client/cas.go

```diff
@@ -975,6 +975,7 @@ func copyFile(srcExecRoot, dstExecRoot, from, to string, mode os.FileMode) error
 type downloadRequest struct {
 	digest   digest.Digest
 	execRoot string
+	context  context.Context
```

Comment:
I tend to try to avoid embedding a context.Context in structs. I guess the other alternative is to monitor a channel? Would that be too complicated if it just covers the case of downloadSingle as you currently do?

Reply:
Yes, I thought it would be needlessly complicated. Instead of a context the request would include a channel, which the client would then ping, and we would create our own context.WithCancel, a goroutine waiting on that channel that would call cancel on the new context (see upload cancellations PR for comparison). I can do all that -- but I thought that would be needlessly complicated. Why not we embed a context on a struct in this case?

Reply:
There's a lot of threads about why the guideline exists, but I think Kaveh's response here is the most appropriate: https://groups.google.com/a/google.com/forum/#!searchin/go-nuts/context$20in$20struct/go-nuts/MAp-5Zt9z7M/qORACKu-AQAJ